app/api/hooks.ts

```diff
@@ -29,17 +26,12 @@ import { invariant } from '~/util/invariant'
 import type { ApiResult } from './__generated__/Api'
 import { processServerError, type ApiError } from './errors'
 import { navToLogin } from './nav-to-login'
-import { type ResultsPage } from './util'
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-export type Params<F> = F extends (p: infer P) => any ? P : never
-export type Result<F> = F extends (p: any) => Promise<ApiResult<infer R>> ? R : never
-export type ResultItem<F> =
-  Result<F> extends { items: (infer R)[] }
-    ? R extends Record<string, unknown>
-      ? R
-      : never
-    : never
```

Comment:
`ResultItem` was only used by the old QueryTable, I think.

Reply:
oof; that's an ugly one